modules/core/src/main/java/com/google/refine/importers/ImporterUtilities.java

```diff
@@ -198,6 +198,26 @@ static public void setupColumns(Project project, List<String> columnNames) {
         project.columnModel.update();
     }
 
+    /**
+     * @param columnsHasData
+     *            Record if there is data in each column( false:null;true:has data)
+     */
+    static public void deleteEmptyColumns(Project project, List<Boolean> columnsHasData) throws ModelException {
+        // skip, if no columns to delete
+        if (columnsHasData.isEmpty() || columnsHasData.stream().allMatch(Boolean.TRUE::equals)) {
+            return;
+        }
+
+        project.columnModel.update(); // make sure all our cell indexes are up to date
+        for (int c = 0; c < columnsHasData.size(); c++) {
+            if (Boolean.FALSE.equals(columnsHasData.get(c))) {
+                // remove column from columns
```

Comment:
```suggestion
            if (!columnsHasData.get(c))) {
```

or `!columnsHasData[c]` with a `boolean[]`

Reply:
thanks for the explanation, but I'm already familiar with the difference between boolean/Boolean and accessing elements in arrays and lists